spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AtomicOperationsCombinedDetector.java

```diff
@@ -0,0 +1,323 @@
+import org.apache.bcel.generic.MONITORENTER;
+import org.apache.bcel.generic.MONITOREXIT;
+import org.apache.bcel.generic.PUTFIELD;
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.LocalVariableAnnotation;
+import edu.umd.cs.findbugs.OpcodeStack;
+import edu.umd.cs.findbugs.ba.CFG;
+import edu.umd.cs.findbugs.ba.CFGBuilderException;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.ClassMember;
+import edu.umd.cs.findbugs.ba.DataflowAnalysisException;
+import edu.umd.cs.findbugs.ba.Location;
+import edu.umd.cs.findbugs.ba.OpcodeStackScanner;
+import edu.umd.cs.findbugs.ba.XFactory;
+import edu.umd.cs.findbugs.ba.XField;
+import edu.umd.cs.findbugs.ba.XMethod;
+import edu.umd.cs.findbugs.ba.ca.Call;
+import edu.umd.cs.findbugs.ba.ca.CallList;
+import edu.umd.cs.findbugs.ba.ca.CallListDataflow;
+import edu.umd.cs.findbugs.bcel.BCELUtil;
+import edu.umd.cs.findbugs.classfile.CheckedAnalysisException;
+import edu.umd.cs.findbugs.classfile.MethodDescriptor;
+import edu.umd.cs.findbugs.util.CollectionAnalysis;
+import edu.umd.cs.findbugs.util.MethodAnalysis;
+
+public class AtomicOperationsCombinedDetector implements Detector {
+
+    private static class BugPrototype {
+        final ClassContext classContext;
+        final Method method;
+        final Location location;
+        XField invokedField;
+        XMethod invokedMethod;
+        LocalVariableAnnotation localVariableAnnotation;
+
+        private BugPrototype(ClassContext classContext, Method method, Location location) {
+            this.classContext = classContext;
+            this.method = method;
+            this.location = location;
+        }
+
+        private BugInstance toBugInstance(Detector detector, String type) {
+            BugInstance bugInstance = new BugInstance(detector, type, LOW_PRIORITY)
+                    .addClassAndMethod(classContext.getJavaClass(), method)
+                    .addSourceLine(classContext, method, location);
+            if (invokedField != null) {
+                bugInstance.addField(invokedField);
+            } else if (localVariableAnnotation != null) {
+                bugInstance.add(localVariableAnnotation);
+            }
+            if (invokedMethod != null) {
+                bugInstance.addCalledMethod(invokedMethod);
+            }
+            return bugInstance;
+        }
+    }
+
+    private final BugReporter bugReporter;
+    private final BugAccumulator bugAccumulator;
+
+    private final Set<XField> interestingFields = new HashSet<>();
+    private final Set<XField> combinedAtomicFields = new HashSet<>();
+
+    private final Map<Method, Map<XField, List<BugPrototype>>> interestingFieldCalls = new HashMap<>();
+    private final Map<Method, List<BugPrototype>> interestingLocalVariableCalls = new HashMap<>();
```

Comment:
What makes a local variable or field interesting? Can you please rename these fields and functions to describe better what does these contain or what the function does?

Reply:
I removed the "interesting" word from variables and methods.